test/pet.test.js

```diff
@@ -1,24 +1,246 @@
+    petCat.hunger = 10;
+    expect(() => petCat.growUp()).toThrow(petCat.deathString);
+  });
+
+  it('Pet cannot grow if fitness is 0', () => {
+    petCat.fitness = 0;
+    expect(() => petCat.growUp()).toThrow(petCat.deathString);
+  });
+
+});
+
+describe('walk', () => {
+  it('takes the pet for a walk and increments fitness level by 4', () => {
+
+    petCat.fitness = 4;
+    petCat.walk();
+    expect(petCat.fitness).toEqual(8);
+  });
+
+  it ('Does not let fitness be more than 10', () => {
+
+    petCat.fitness = 8;
+    petCat.walk();
+    expect(petCat.fitness).toEqual(10);
+
+  });
+
+  it('Pet cannot walk if fitness is 0', () => {
+    petCat.fitness = 0;
+    expect(() => petCat.walk()).toThrow(petCat.deathString);
+
+});
+});
+
+describe('stats', () => {
+  it('returns a string of the current stats of petCat', () => {
+
+    petCat.age = 1;
+    petCat.fitness = 10;
+    petCat.hunger = 5;
+    petCat.stats();
+    
+    expect(petCat.statsString).toBe('Pet Stats: NAME: Rocky, AGE: 1, FITNESS: 10, HUNGER: 5');   
+
+  });
+});
+
+
+describe('feed', () => {
+  it('feeds petCat and decrements hunger level by 3', () => {
+
+    petCat.hunger = 9;
+    petCat.feed();
+    expect(petCat.hunger).toEqual(6);
+
+  });
 
-  it('GROW UP: increments the age by 1 & hunger by 5', () => {
+  it('Does not let hunger go beyond 0', () => {
+
+    petCat.hunger = 1;
+    petCat.feed();
+    expect(petCat.hunger).toEqual(0);
+
+  });
+
+  it('Pet cannot be fed if hunger is 10', () => {
+    petCat.hunger = 10;
+    expect(() => petCat.feed()).toThrow(petCat.deathString);
+  
+  });
+});
+
+
+describe('checkUp', () => {
+  it('returns PetCat is hungry & needs a walk' , () => {
+    petCat.fitness = 1;
+    petCat.hunger = 9;
+    petCat.checkUp();
+  
+    expect(petCat.checkUp()).toBe('I am hungry AND I need a walk');
+  });
+
+  it('returns PetCat just needs a walk' , () => {
+    petCat.fitness = 1;
+    petCat.hunger = 3;
+    petCat.checkUp();
+  
+    expect(petCat.checkUp()).toBe('I need a walk');
+
+  });
+
+  it('returns PetCat just is hungry' , () => {
+    petCat.fitness = 9;
+    petCat.hunger = 8;
+    petCat.checkUp();
+
+    expect(petCat.checkUp()).toBe('I am hungry');
+
+  });
+
+  it('returns PetCat feels great' , () => {
+    petCat.fitness = 9;
+    petCat.hunger = 2;
+    petCat.checkUp();
+
+    expect(petCat.checkUp()).toBe('I feel great!');
+
+  });
+
+  it('returns deathString if pet is dead' , () => {
+
+    petCat.age = 30;
+    petCat.hunger = 10;
+    petCat.fitness = 0;
+    expect(() => petCat.checkUp()).toThrow(petCat.deathString);
+
+  });
+
+});
+describe('isAlive', () => {
+  it ('returns true for age < 30 or false for age > 30', () => {
+    petCat.age = 7;
+    expect(petCat.isAlive).toBe(true);
+
+
+    petCat.age = 30;
+    expect(petCat.isAlive).toBe(false);
+    expect(() => petCat.deathChecker()).toThrow(petCat.deathString);
+
+  });
+
+  it ('returns true for hunger < 10 or false for hunger > 10', () => {
```

Comment:
I would probably separate these so in future if one failed, it would be immediately obvious which one, the true or the false check.